src/m365/spo/commands/tenant/tenant-applicationcustomizer-remove.ts

```diff
@@ -0,0 +1,187 @@
+      });
+
+      if (result.continue) {
+        await this.removeTenantApplicationCustomizer(logger, args);
+      }
+    }
+  }
+
+  public async getTenantApplicationCustomizer(logger: Logger, args: CommandArgs, requestUrl: string): Promise<number> {
+    if (this.verbose) {
+      logger.logToStderr(`Getting the tenant application customizer ${args.options.id || args.options.title || args.options.clientSideComponentId}`);
+    }
+
+    const filter: string[] = [`TenantWideExtensionLocation eq 'ClientSideExtension.ApplicationCustomizer'`];
+    if (args.options.title) {
+      filter.push(`Title eq '${args.options.title}'`);
+    }
+    else if (args.options.id) {
+      filter.push(`Id eq '${args.options.id}'`);
+    }
+    else if (args.options.clientSideComponentId) {
+      filter.push(`TenantWideExtensionComponentId eq '${args.options.clientSideComponentId}'`);
+    }
+
+    const reqOptions: CliRequestOptions = {
+      url: `${requestUrl}/items?$filter=${filter.join(' and ')}`,
+      headers: {
+        'accept': 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+
+    const listItemInstances: ListItemInstanceCollection = await request.get<ListItemInstanceCollection>(reqOptions);
+
+    if (listItemInstances.value.length === 0) {
+      throw 'The specified application customizer was not found';
+    }
+
+    if (listItemInstances.value.length > 1) {
+      throw `Multiple application customizers with ${args.options.title || args.options.clientSideComponentId} were found. Please disambiguate (IDs): ${listItemInstances.value.map(item => item.Id).join(', ')}`;
+    }
+
+    return listItemInstances.value[0].Id;
+
+  }
+
+  private async removeTenantApplicationCustomizer(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      if (this.verbose) {
+        logger.logToStderr(`Removing tenant application customizer ${args.options.id || args.options.title || args.options.clientSideComponentId}`);
+      }
+
+      const appCatalogUrl = await spo.getTenantAppCatalogUrl(logger, this.debug);
+      if (!appCatalogUrl) {
+        throw 'No app catalog URL found';
+      }
+
+      const listServerRelativeUrl: string = urlUtil.getServerRelativePath(appCatalogUrl, '/lists/TenantWideExtensions');
+
+      const requestUrl = `${appCatalogUrl}/_api/web/GetList('${formatting.encodeQueryParameter(listServerRelativeUrl)}')`;
+
+      const id = await this.getTenantApplicationCustomizer(logger, args, requestUrl);
+
+      const requestOptions: CliRequestOptions = {
+        url: `${requestUrl}/items(${id})`,
+        method: 'POST',
+        headers: {
+          'X-HTTP-Method': 'DELETE',
+          'If-Match': '*',
+          'accept': 'application/json;odata=nometadata'
+        },
+        responseType: 'json'
+      };
+
+      await request.post(requestOptions);
+    }
+    catch (err: any) {
+      logger.log(err);
```

Comment:
The handle function below should take care of the logging part. So this line is superfluous.